src/services/WolfDen.Application/Requests/Commands/Attendence/StatusUpdation/StatusUpdateCommandHandler.cs

```diff
@@ -0,0 +1,138 @@
+﻿using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using WolfDen.Application.Requests.Commands.Attendence.Email;
+using WolfDen.Domain.Entity;
+using WolfDen.Infrastructure.Data;
+
+namespace WolfDen.Application.Requests.Commands.Attendence.StatusUpdation
+{
+    public class StatusUpdateCommandHandler:IRequestHandler<StatusUpdateCommand,int>
+    {
+        private readonly WolfDenContext _context;
+        private readonly string SenderEmail;
+        private readonly string SenderName;
+        public StatusUpdateCommandHandler(WolfDenContext context, IConfiguration configuration)
+        {
+            _context = context;
+            SenderEmail = configuration["BrevoApi:SenderEmail"];
+            SenderName = configuration["BrevoApi:SenderName"];
+
+        }
+        public async Task<int> Handle(StatusUpdateCommand request, CancellationToken cancellationToken)
+        {
+            int statusId = 0;
+            var day = await _context.DailyAttendence.Where(x => x.EmployeeId == request.EmployeeId && x.Date == request.Date).FirstOrDefaultAsync(cancellationToken);
+            if (day == null)
+            {
+                var holiday = await _context.Holiday.Where(x => x.Date == request.Date).FirstOrDefaultAsync();
+              
+                    if (holiday.Type == "Normal")
+                    {
+                        statusId = 5;
+                    }
+                    else
+                    {
+                        //var leave = await _context.LeaveRequest.Where(x => x.EmpId == request.EmployeeId && x.FromDate == request.Date && x.status == 2).FirstOrDefaultAsync(cancellationToken);
+                        //if (leave != null)
+                        //{
+                        //    if (leave.TypeName == "WFH")
+                        //    {
+                        //        statusId = 3;
+                        //    }
+                        //    else
+                        //    {
+                                statusId = 5;
+                        //    }
+                        //}
+                    }
+
+                
+            }
+            else
+            {
+                var insideHrs = await _context.DailyAttendence.Where(x => x.EmployeeId == request.EmployeeId && x.Date == request.Date).Select(x => x.InsideDuration).FirstOrDefaultAsync(cancellationToken);
+                if (insideHrs < 360)
+                {
+                    statusId = 2;

```

Comment:
what is is 2 ,1 ,5. Are these enum values? If so pass as enum.

https://stackoverflow.com/questions/47882/what-are-magic-numbers-and-why-do-some-consider-them-bad

Reply:
not enum,it is foreignkey from the table StatusType ,which contains status like present,absent,late...